packages/components/src/core/ComplexFilter/index.tsx

```diff
@@ -69,19 +69,23 @@ const ComplexFilter = <
   const [open, setOpen] = useState(false);
   const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null);
 
-  const [value, setValue] = useState(getInitialValue());
+  const [value, setValueState] = useState(getInitialValue());
   const [pendingValue, setPendingValue] = useState(getInitialValue());
 
-  useEffect(() => {
-    onChange(value);
-    setPendingValue(value);
-  }, [onChange, value]);
+  const setValue = useCallback(
+    (nextValue: AutocompleteValue<T, Multiple, DisableClearable, FreeSolo>) => {
+      setValueState(nextValue);
+      onChange(value);
+      setPendingValue(nextValue);
+    },
+    [onChange, value]
+  );
```

Comment:
I also wonder we should compare the existing value in state before calling onChange too, in case the component user passes a new arrow function at each render and the value actually didn't change?

E.g.,

```tsx
<ComplexFilter onChange={() => { ... }} />
```

Something like:

```tsx
const prevValueRef = useRef();

  useEffect(() => {
    if (prevValueRef.current !== value) {
      onChange(value);
      setPendingValue(value);
      prevValueRef.current = value;
    }
  }, [value, onChange]);
```